Writing labels into scores with ms3 1.2.0

johentsch opened the pull request:
Hi Andrew, 

with this suggestion, utils.eval_utils.write_labels_to_score() is not needed anymore. write_to_score.write_tsvs_to_scores() on the other hand now has various options for where and how to insert the labels. If you prefer fixing them, we can simply set them to the defaults used before. That's why I haven't integrated them in the CLI.

Was it done on purpose that previously the annotated TSV files would be named only with the last word of the original file name? Because this would not work with ms3's file matching which is entirely based on the idea of a fixed `fname` component plus arbitrary suffixes. This is why I made the little change in annotate.py, hope you're OK with that.

Commit:
ensures original file names for annotation files

Commit:
uses score.mscx.labels()

Commit:
makes use of ms3 1.2.0 in write_to_score.write_tsvs_to_scores()

apmcleod commented on annotate.py:
```diff
@@ -203,8 +203,8 @@ def annotate(
             )
 
             if piece.name is not None and output_tsv_dir is not None:
-                piece_name = Path(piece.name.split(" ")[-1])
```
Please edit this line to the following 2 lines:

```
# Removes the ID and first space from piece.name, and stems remaining path
piece_name = Path(" ".join(piece.name.split(" ")[1:])).stem
```

The idea is just to remove what's before the first space, and then stem. You're version would cause an error if there is no directory in piece.name.

johentsch replied:
Did that in [7da775f](https://github.com/apmcleod/harmonic-inference/pull/73/commits/7da775f4d1e144d380b49b324273abbc11e1b527). However, when I last used annotate with the previous `piece_name = Path(piece.name).stem`, the output TSVs did not include any IDs.

> Your version would cause an error if there is no directory in piece.name.

I don't see this, what would be an example `piece.name` that would yield an error?

apmcleod replied:
I think for the ones you were testing, `piece.name` looked like `123: Bach/op12.msc`, where stem removes everything before the `/` and after the last `.`. I think it would error on something like `123: op12.msc`, although I'm not sure exactly what combination of args to `aggregate_corpus_data.py` would produce that (or if that's even possible, though I suspect it is).

apmcleod commented on harmonic_inference/utils/forces.py:
```diff
@@ -223,7 +223,7 @@ def decode_key(
 
     score = Score(score_path)
-    labels: pd.DataFrame = score.annotations.get_labels(decode=True)
+    labels: pd.DataFrame = score.mscx.labels()
```
I didn't test this yet.

johentsch replied:
It does the same thing but through the default interface (that would also allow unfolding repeats and adding an interval_index).

apmcleod commented on write_to_score.py:
```diff
@@ -31,29 +33,76 @@ def write_tsvs_to_scores(
         The path to annotations and MuseScore3 scores, whose sub-directories and file names
         are aligned with those in the output TSV directory.
 
-    fuzzy : bool
```
This now causes an error in the script. Please also remove the --fuzzy arg and its inclusion in the call to this function.

johentsch replied:
[3c57d40](https://github.com/apmcleod/harmonic-inference/pull/73/commits/3c57d40ef6dbe84a6ddf85c34254e06756f872f2)

apmcleod commented on write_to_score.py:
```diff
@@ -31,29 +33,76 @@ def write_tsvs_to_scores(
+        # if the Corpus directory itself includes any labels we exclude them to avoid ambiguity
+        if any(x.is_dir() and x.name == "labels" for x in annotations_base_dir.iterdir()):
+            excluded_labels_path = annotations_base_dir / "labels"
+        corpus.view.exclude("path", str(excluded_labels_path))
```
Error here if the above if is False. Probably this just needs to be indented?

johentsch replied:
Indeed, sorry

apmcleod reviewed (changes requested):
I wasn't able to test the actual write to score yet. When I run it with -o outputs --annotations all_subcorpora (as it was working before), I get many messages and prompts from ms3 about irrelevant files.

E.g.:
```
17 pieces require disambiguation for facet 'scores'.
(1/2)  Disambiguation for the following 3 pieces (17 remaining) by choosing one of the 2 columns:

                                 scores, fext: .mscx  scores, fext: .mscz
Snarky_Puppy-Shapons_Vindaloo                   3056                 3057
The_Fearless_Flyers-Ace_of_Aces                 3060                 3061
Vulfpeck-Fugue_State                            3064                 3065

Choose one of the columns (or 0 for none):
1 -> scores, fext: .mscx
2 -> scores, fext: .mscz
Selection [1-2]: 
```

Here, my annotations didn't include any of those files, so they shouldn't be needed, I would have thought.

This functionality is important, since outputs may contain annotations for many subcorpora, which would necessitate passing the all_subcorpora path, rather than any specific subcorpus.

In terms of the other arguments, if there isn't a current use case, I would probably leave them all as default, and add them back in (to the write_tsvs_to_scores function _and_ to the CLI as needed.

I've also noticed an unused function write_tsvs_to_scores in test.py:30. Can you please remove it as well?

Commit:
removes --fuzzy option from write_to_score.py

Commit:
write_tsvs_to_scores() uses filter_other_fnames=True

ms3 1.2.1 introduces this argument to allow for parsing only scores that correspond to a file in the added directory

Commit:
copies the new write_tsvs_to_scores() to test.py

Commit:
removes ID from piece.name

johentsch commented:
Thanks for the review, @apmcleod , and sorry for the half-baked PR. The added commits should solve the described problems.

apmcleod commented:
Thanks! I'll test it again tomorrow, but it looks like everything should be fine now.
